src/ray/core_worker/core_worker_process.cc

```diff
@@ -0,0 +1,361 @@
+// Copyright 2017 The Ray Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "ray/core_worker/core_worker.h"
+#include "ray/stats/stats.h"
+#include "ray/util/event.h"
+#include "ray/util/util.h"
+
+namespace ray {
+namespace core {
+namespace {
```

Comment:
What's this empty namespace for?

Reply:
https://en.cppreference.com/w/cpp/language/namespace#Unnamed_namespaces
TL;DR: anonymous namespace is away to avoid ODR issues by defining names local to the current translation unit, similar to what `static` does. This is a [preferred way over static](https://techmunching.com/unnamed-anonymous-namespaces-vs-namespace-static/).